### DISCUSSION TITLE
"sudo: !!: command not found"

### DISCUSSION THREAD

**Discussion by cicu812** [leo-arch/clifm#243] [Ideas]
Hi. Not such a big deal most of the time, but I'll have to bounce out of clifm if I wanna do any bang bang because: 

> sudo: !!: command not found

**Comment by leo-arch** [comment]
Hi @cicu812. Could you please paste here the exact command you're trying to run? Also, what version of **clifm** are you running?

EDIT: Ok, get it. It's just `sudo !!` isn't it?

**Reply by cicu812** [reply]
I bounced out... Can't remember exactly what I was doing. I think I was editing a remote dot file. I'm new to the tool but I love it so far. Thank you. sudo does work for anything in clifm, but if I have a permission denied error, bang bang is convenient for whatever that last command was. I'm using version 1.13.1. clifm does not report the version in app:

`<0> $ clifm ---version 
clifm: To launch a new instance use the 'x' command`

**Comment by leo-arch** [comment]
Yes, it isn't working. For the time being, you can use this alternative procedure: press <kbd>Up</kbd> (to invoke the last executed command), and then <kbd>Alt-v</kbd> to insert the `sudo` command.

**Reply by cicu812** [reply]
Awesome. Thank you.

**Reply by leo-arch** [reply]
However, `sudo !!` should work (though it isn't working right now). Added to my TODO list. Thanks for pointing it out.

**Reply by cicu812** [reply]
Appreciate your work and your valuable time. 

All the best to you. I'll give you the best feedback I can.

**Comment by leo-arch** [comment]
To check **clifm**'s version in app just run `ver` or `version`.

**Comment by cicu812** [comment]
So I see this tip: "!! to repeat the last command." And I"m thinking what  a nice guy Leo is. haha

I go to test it to see if it's any different from my last experience. 
# here's a protected file
`vim /etc/apt/sources.list`
# let's say I decide to edit, knowing it's rendered "read only". 
# I quit vim, back in clifm, I sudo bang bang to repeat the last command elevated
<0> $ sudo !!
sudo: !!: command not found

Oops. bang bang say sudo is on sabatical. Try back again later like maybe when Leo gets in, or type it all out. Don't be lazy like you are on the command line. This is clifm for g'sake. Sit up straight. No slouching! ;)

**Reply by leo-arch** [reply]
Done. There are still some rough edges, but it basically works.

**Reply by leo-arch** [reply]
Hi @cicu812. Have you tried this fix?

**Reply by cicu812** [reply]
I love it! I'm sorry it took me a minute to get back to you, but I didn't forget. THANK YOU SO MUCH! You're awesome! Works perfect!
